packages/kamino-sdk/src/Kamino.ts

```diff
@@ -250,6 +272,151 @@ export class Kamino {
+      throw new Error(`Rebalance method ${rebalanceMethod} is not supported`);
+    }
+  };
+
+  getFieldsForManualRebalanceMethod = async (
+    dex: Dex,
+    fieldOverrides: RebalanceFieldInfo[],
+    tokenAMint: PublicKey,
+    tokenBMint: PublicKey
+  ): Promise<RebalanceFieldInfo[]> => {
+    let price = await this.getPriceForPair(dex, tokenAMint, tokenBMint);
+
+    let lowerPrice: number;
+    let lowerPriceInput = fieldOverrides.find((x) => x.label == 'lowerPrice');
+    if (lowerPriceInput) {
+      lowerPrice = lowerPriceInput.value;
+    } else {
+      lowerPrice = (price * (FullBPS - DefaultLowerPriceDifferenceBPS)) / FullBPS;
+    }
+
+    let upperPrice: number;
+    let upperPriceInput = fieldOverrides.find((x) => x.label == 'upperPrice');
+    if (upperPriceInput) {
+      upperPrice = upperPriceInput.value;
+    } else {
+      upperPrice = (price * (FullBPS + DefaultUpperPriceDifferenceBPS)) / FullBPS;
+    }
+
+    return getManualRebalanceFieldInfos(lowerPrice, upperPrice);
+  };
+
+  getFieldsForPricePercentageMethod = async (
+    dex: Dex,
+    fieldOverrides: RebalanceFieldInfo[],
+    tokenAMint: PublicKey,
+    tokenBMint: PublicKey
+  ) => {
+    let price = await this.getPriceForPair(dex, tokenAMint, tokenBMint);
+
+    let lowerPriceDifferenceBPS: number;
+    let lowerPriceDifferenceBPSInput = fieldOverrides.find((x) => x.label == 'lowerThresholdBps');
+    if (lowerPriceDifferenceBPSInput) {
+      lowerPriceDifferenceBPS = lowerPriceDifferenceBPSInput.value;
+    } else {
+      lowerPriceDifferenceBPS = DefaultLowerPriceDifferenceBPS;
+    }
+
+    let upperPriceDifferenceBPS: number;
+    let upperPriceDifferenceBPSInput = fieldOverrides.find((x) => x.label == 'upperThresholdBps');
+    if (upperPriceDifferenceBPSInput) {
+      upperPriceDifferenceBPS = upperPriceDifferenceBPSInput.value;
+    } else {
+      upperPriceDifferenceBPS = DefaultUpperPriceDifferenceBPS;
+    }
+
+    let lowerPrice = (price * (FullBPS - lowerPriceDifferenceBPS)) / FullBPS;
+    let upperPrice = (price * (FullBPS + upperPriceDifferenceBPS)) / FullBPS;
+    let fieldInfos = getPricePercentageRebalanceFieldInfos(lowerPriceDifferenceBPS, upperPriceDifferenceBPS).concat(
+      getManualRebalanceFieldInfos(lowerPrice, upperPrice, false)
+    );
+
+    return fieldInfos;
+  };
+
+  getPriceForPair = async (dex: Dex, poolTokenA: PublicKey, poolTokenB: PublicKey): Promise<number> => {
+    if (dex == 'ORCA') {
+      let pools = await this.getOrcaPoolsForTokens(poolTokenA, poolTokenB);
+      if (pools.length == 0) {
+        throw new Error(`No pool found for ${poolTokenA.toString()} and ${poolTokenB.toString()}`);
+      }
+      return pools[0].price;
+    } else if (dex == 'RAYDIUM') {
+      let pools = await this.getRaydiumPoolsForTokens(poolTokenA, poolTokenB);
+      if (pools.length == 0) {
+        throw new Error(`No pool found for ${poolTokenA.toString()} and ${poolTokenB.toString()}`);
+      }
+      return pools[0].price;
```

Comment:
does this return the a/b price or the pool of price which could be b/a?

Reply:
yes. I will make it private in the next PR and add comment explaining that it just reads the price from the pool